ksqldb-engine/src/test/java/io/confluent/ksql/engine/RuntimeAssignorTest.java

```diff
@@ -0,0 +1,100 @@
+package io.confluent.ksql.engine;
+
+import io.confluent.ksql.name.SourceName;
+import io.confluent.ksql.query.QueryId;
+import io.confluent.ksql.util.BinPackedPersistentQueryMetadataImpl;
+import io.confluent.ksql.util.KsqlConfig;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RuntimeAssignorTest {
+
+  @Mock
+  public BinPackedPersistentQueryMetadataImpl queryMetadata;
+  private final QueryId query1 = new QueryId("Test1");
+  private final QueryId query2 = new QueryId("Test2");
+  private final Collection<SourceName> sources1 = Collections.singleton(SourceName.of("test1"));
+  private final Collection<SourceName> sources2 = Collections.singleton(SourceName.of("test2"));
+
+  private String firstRuntime;
+  private RuntimeAssignor runtimeAssignor;
+  private static final KsqlConfig KSQL_CONFIG = new KsqlConfig(Collections.emptyMap());
+
+  @Before
+  public void setUp() {
+    runtimeAssignor = new RuntimeAssignor(KSQL_CONFIG, 2);
+    firstRuntime = runtimeAssignor.getRuntime(
+        query1,
+        sources1,
+        KSQL_CONFIG
+    );
+    when(queryMetadata.getQueryApplicationId()).thenReturn(firstRuntime);
+    when(queryMetadata.getQueryId()).thenReturn(query1);
+    when(queryMetadata.getSourceNames()).thenReturn(new HashSet<>(sources1));
+  }
+
+  @Test
+  public void shouldCreateSandboxAndDroppingAQueryWillNotChangeReal() {
```

Comment:
add similar tests for adding a new runtime, or adding a query to an existing runtime

Reply:
`shouldCreateSandboxAndAddingAQueryAndRuntimeWillNotChangeReal` and `shouldCreateSandboxAndAddingAQueryWillNotChangeReal`